src/app/CommandHandler.cpp

```diff
@@ -82,67 +85,56 @@ CHIP_ERROR CommandHandler::OnInvokeCommandRequest(Messaging::ExchangeContext * e
     // response too.  Figure out at what point it's our responsibility to
     // handler errors vs our caller's.
     Handle workHandle(this);
-    mExchangeCtx->WillSendMessage();
-    ReturnErrorOnFailure(ProcessInvokeRequest(std::move(payload), isTimedInvoke));
 
-    return CHIP_NO_ERROR;
+    mExchangeCtx->WillSendMessage();
+    status = ProcessInvokeRequest(std::move(payload), isTimedInvoke);
+    if (status != Status::Success)
+    {
+        StatusResponse::Send(status, mExchangeCtx.Get(), false /*aExpectResponse*/);
+        mSentStatusResponse = true;
```

Comment:
Shouldn't that get set true only if the Send() call succeeded?  Maybe it doesn't matter much in practice....

Reply:
yes, it does not matter in practice.